pkg/taskrunmetrics/metrics_test.go

```diff
@@ -64,16 +64,18 @@ func TestUninitializedMetrics(t *testing.T) {
 		Status: corev1.ConditionUnknown,
 	}
 
-	if err := metrics.DurationAndCount(&v1beta1.TaskRun{}, beforeCondition); err == nil {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
just curious, what's the need for the cancellation function here?

Reply:
Make sure to teardown the resources.